log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationFactory.java

```diff
@@ -355,28 +348,9 @@ static String extractClassLoaderUriPath(final URI uri) {
      * @param loader The default ClassLoader to use.
      * @return The InputSource to use to read the configuration.
      */
+    @Deprecated
     protected ConfigurationSource getInputFromString(final String config, final ClassLoader loader) {
-        try {
-            final URL url = new URL(config);
-            URLConnection urlConnection = UrlConnectionFactory.createConnection(url);
-            File file = FileUtils.fileFromUri(url.toURI());
-            if (file != null) {
-                return new ConfigurationSource(urlConnection.getInputStream(), FileUtils.fileFromUri(url.toURI()));
-            }
-            return new ConfigurationSource(urlConnection.getInputStream(), url, urlConnection.getLastModified());
-        } catch (final Exception ex) {
-            final ConfigurationSource source = ConfigurationSource.fromResource(config, loader);
-            if (source == null) {
-                try {
-                    final File file = new File(config);
-                    return new ConfigurationSource(new FileInputStream(file), file);
-                } catch (final FileNotFoundException fnfe) {
-                    // Ignore the exception
-                    LOGGER.catching(Level.DEBUG, fnfe);
-                }
-            }
-            return source;
-        }
+        return ConfigurationSource.fromUri(NetUtils.toURI(config));
```

Comment:
This method has one call site and is called with the context classloader. Ignoring one parameter should be Ok.